[iOS][GPUP] Only block unused syscalls on newer OS versions

pvollan opened the pull request:
#### 45b7e1d0db7c98de18933e830aa1068b0000f6d9
<pre>
[iOS][GPUP] Only block unused syscalls on newer OS versions
<a href="https://bugs.webkit.org/show_bug.cgi?id=241045">https://bugs.webkit.org/show_bug.cgi?id=241045</a>

Reviewed by NOBODY (OOPS!).

Only block unused syscalls on newer OS versions, since telemetry has not been inspected for older OS versions,
and since this version of the sandbox will in general not be applied on older OS versions.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:
</pre>

geoffreygaren commented on Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.h:
```diff
@@ -61,6 +61,9 @@ class RemoteImageBuffer : public WebCore::ConcreteImageBuffer<BackendType> {
     {
+        // Volatile image buffers do not have contexts.
+        if (this->volatilityState() == WebCore::VolatilityState::Volatile)
+            return;
```
This change doesn't look related to the sandboxing change described in the commit log. Is it accidental?

pvollan replied:
Ah, yes, good catch! Removed in latest patch.

geoffreygaren commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:
```diff
@@ -714,7 +714,11 @@
 )
 
 (when (defined? 'syscall-unix)
+#if !PLATFORM(IOS) || __IPHONE_OS_VERSION_MIN_REQUIRED >= 160000
```
This isn't the first case in this file, but I wonder: Why do we test PLATFORM(IOS) in the iOS sandbox file? Isn't it always true?

pvollan replied:
That is a good point. We share this sandbox among all OS's in the iOS family, and this version check is only valid for iOS.

geoffreygaren commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:
```diff
@@ -714,7 +714,11 @@
 (when (defined? 'syscall-unix)
+#if !PLATFORM(IOS) || __IPHONE_OS_VERSION_MIN_REQUIRED >= 160000
     (deny syscall-unix (with telemetry))
+#else
+    (allow syscall-unix)
+#endif
```
I guess this is just to enable Apple-internal testing of newer iOS WebKit builds on older OS's? Seems OK to do that, since we also maintain testers on the newer OS's.

pvollan replied:
Yes, this is not a rule that will be applied on newer OS versions.

geoffreygaren reviewed (changes requested):
Requesting changes to remove the volatility state stuff if it was by accident, or else explain it in the commit message if it was on purpose.

pvollan committed:
[iOS][GPUP] Only block unused syscalls on newer OS versions
https://bugs.webkit.org/show_bug.cgi?id=241045

Reviewed by NOBODY (OOPS!).

Only block unused syscalls on newer OS versions, since telemetry has not been inspected for older OS versions,
and since this version of the sandbox will in general not be applied on older OS versions.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:
